### PR TITLE
Rails 3.0 support

### LINKED ISSUE #13: Should this work with Rails 3.0?
When trying to use in an application I'm upgrading from 2.3.18 to 3.0.20 I'm getting an uninitialized constant ActionDispatch::Flash::KEY in lib/rails_4_session_flash_backport/rails3/flash_hash.rb:58:in `ensure in call' while on Rails 3.0.20.  This is with v0.1.0 of rails_4_session_flash_backport.

It doesn't look like ActionDispatch::Flash::KEY is defined in 3.0, just in 3.1 and 3.2.

### PR DESCRIPTION
**Pull request by sj26**
Adds rails 3.0 support and clarifies the caveats in the README. Fixes #13.

If you're already on rails 3.0 and upgrade straight to 3.1 then your sessions are going to [break with ArgumentError (dump format error) anyway](http://stackoverflow.com/questions/9120501/what-causes-the-argumenterror-dump-format-error). Installing this gem first actually works around this by storing flash objects more simply. Installing it back on 2.3 and upgrading through 3.0 to 3.1+ should see no issues.

/cc @johnsyweb @joneslee85 @ryanpool

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -12,31 +12,51 @@ crazy stuff we might as well go and use the far more sensible practice from
...
+
+### Rails 3.0
+
+Rails 3.0 was a weird half-way house between 2.x and 3.1. The 3.0 to 3.1
+upgrade gave many of us an [Argument Error (dump format error)
+problem][argumenterror-so].
+
+    [argumenterror-so]: http://stackoverflow.com/questions/9120501/what-causes-the-argumenterror-dump-format-error
+
+On Rails 3.0, rails 3.0 flashes can be successfully decoded, but the other
```
Comment (joneslee85) [inline README.md]: s/rails 3.0//

**Comment by joneslee85** [comment]
:+1: :clap: :100:

**Comment by johnsyweb** [comment]
:shipit:

File: README.md (hunk 1 above)
Comment (sj26) [inline README.md]: No this means that when running in rails 3.0 then a flash encoded by vanilla rails 3.0 can be decoded, but if that wasn't clear then I should reword this sentence.

**Pull request merged**